platforms/jvm/testing-jvm/src/main/java/org/gradle/api/tasks/testing/junitplatform/JUnitPlatformOptions.java

```diff
@@ -16,43 +16,28 @@
 
 package org.gradle.api.tasks.testing.junitplatform;
 
+import org.gradle.api.provider.SetProperty;
 import org.gradle.api.tasks.Input;
 import org.gradle.api.tasks.testing.TestFrameworkOptions;
-import org.gradle.internal.instrumentation.api.annotations.ToBeReplacedByLazyProperty;
-
-import java.util.Arrays;
-import java.util.LinkedHashSet;
-import java.util.Set;
+import org.gradle.internal.instrumentation.api.annotations.ReplacesEagerProperty;
 
 /**
  * The JUnit platform specific test options.
  *
  * @see <a href="https://junit.org/junit5/docs/current/user-guide">JUnit 5 User Guide</a>
  * @since 4.6
  */
-public class JUnitPlatformOptions extends TestFrameworkOptions {
-    private Set<String> includeEngines = new LinkedHashSet<String>();
-
-    private Set<String> excludeEngines = new LinkedHashSet<String>();
-
-    private Set<String> includeTags = new LinkedHashSet<String>();
-
-    private Set<String> excludeTags = new LinkedHashSet<String>();
+public abstract class JUnitPlatformOptions extends TestFrameworkOptions {
```

Comment:
❓ Doesn't making this class `abstract` count as a breaking change to the API that has to wait for 9.0?

Same with `JUnitOptions`.

Reply:
Yes, this is a breaking change. But it will be only merged in to Provider API EAP branch.

Reply:
I've opened https://github.com/gradle/gradle/issues/30373 so we don't forget to document it